fv3_cap.F90

```diff
@@ -0,0 +1,1691 @@
+    read (io_unit,*,err=999) calendar_type
+    read (io_unit,*) date_init
+    read (io_unit,*) date
+    close(io_unit)
+    force_date_from_configure = .false.
+!
+    if(date(1) == 0 .and. date_init(1) /= 0) date = date_init
+    if(mype == 0) print *,'bf clock_fv3,date=',date,'date_init=',date_init
+
+    call ESMF_VMbroadcast(vm, date, 6, 0)
+    call ESMF_TimeSet(time=CurrTime,yy=date(1),mm=date(2),dd=date(3),h=date(4), &
+                      m=date(5),s=date(6),rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+999 continue
+998 continue
+!    if(mype==0) print *,'final date =',date,'date_init=',date_init
+
+!reset CurrTime in clock
+    call ESMF_ClockSet(clock_fv3, currTIME=CurrTime, startTime=startTime,  &
+                       stopTime=stopTime, timeStep=timeStep, rc=rc)
+
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+ 
+    !
+    !Under NUOPC, the EARTH driver clock is a separate instance from the
+    ! - fv3 clock. However, the fv3 clock may have been reset from restart
+    ! - therefore the EARTH driver clock must also be adjusted.
+    ! - Affected: currTime, timeStep
+    call ESMF_ClockGet(clock, timeStep=earthStep, rc=RC)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
+      line=__LINE__, file=__FILE__)) call ESMF_Finalize(endflag=ESMF_END_ABORT)
+
+    if (earthStep > (stopTime-currTime)) earthStep = stopTime - currTime
+    call ESMF_ClockSet(clock, currTime=currTime, timeStep=earthStep, rc=RC)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
+      line=__LINE__, file=__FILE__)) call ESMF_Finalize(endflag=ESMF_END_ABORT)
+
+    ! Set fv3 component clock as copy of EARTH clock.
+    call NUOPC_CompSetClock(gcomp, clock, rc=RC)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
+      line=__LINE__, file=__FILE__)) call ESMF_Finalize(endflag=ESMF_END_ABORT)
+
+    ! Read in the FV3 coupling interval
+    if ( cpl ) then
+      call clock_cplIntval(gcomp, CF)
+    endif
+!
+    first_kdt = 1
+    if( output_1st_tstep_rst) then
+      rsthour   = CurrTime - StartTime
+      first_kdt = nint(rsthour/timeStep) + 1
+    endif
+
+!
+!#######################################################################
+! set up fcst grid component
+!
+!----------------------------------------------------------------------
+!*** create fv3 atm tasks and quilt servers
+!-----------------------------------------------------------------------
+!
+! create fcst grid component
+
+    fcstpe = .false.
+    if( quilting ) then
+      num_pes_fcst = petcount - write_groups * wrttasks_per_group
+    else
+      num_pes_fcst = petcount
+    endif
+    allocate(fcstPetList(num_pes_fcst))
+    do j=1, num_pes_fcst
+      fcstPetList(j) = j - 1
+      if(mype == fcstPetlist(j)) fcstpe = .true.
```

Comment:
fcstpe not used in this subroutine
Can be replaced with `ESMF_GridCompIsPetLocal - Inquire if this GridComp is to execute on the calling PET` if needed, although this is usually not needed if calls are structured properly, i.e., calls needed on the fcst pets should be inside the fcst grid comp so the check is not needed.

Reply:
I think fcstpe was set up for some debugging purpose, it can be removed